### DISCUSSION TITLE
Does Puma support interrupting processing of a cancelled request?

### DISCUSSION THREAD

**Discussion by grncdr** [puma/puma#3398] [Q&A] [answered]
Not sure if I'm using the right wording here. I have an API server for a mobile frontend with some endpoints that can be a bit compute/DB intensive. The frontend implements cancellation of requests (e.g. on navigation to a different screen) by closing the connection, but this seems to have no effect on Puma worker threads. Instead the worker thread will finish processing the request for a client that is no longer interested.

What I'm hoping for is some way to raise an exception in the worker thread if the connection is closed during processing. Is that already possible?

**Comment by dentarg** [comment, marked as answer]
I don't think that's possible today
